R/util_data_summaries.R

```diff
@@ -77,17 +69,54 @@ ds.helper$default_col_funcs = list(
 ds.helper$default_categorical_col_funcs = list(
     "pct_NA" = ds.helper$pct_blank,
     "n_unique" = ds.helper$n_unique
-)   
+)  
 
+ds.helper$variable_type <- function(x){
+    # return variable type as
+    # (Note that NAs are ignored)
+    #  boolean - x contains exactly c(0,1) or c(TRUE,FALSE)
+    #  numeric - x is numbers 
+    #  categorical - x is a string or factor with up to 20 levels
+    #  invariant - only 0 or 1 values, not counting NAs
+    
+    boolean <- all( x %in% c(0,1,NA) ) & all( c(0,1) %in% x )
+    
+    if( boolean ){
+        return("boolean")
+    } else if( util.is_vector_of_numbers(x) ){
+        return("numeric")
+    }
+    
+    n_unique <- x[!util.is_blank(x)] %>%
+        unique() %>%
+        length()
+    
+    if(n_unique < 2){
+        return("invariant")
```

Comment:
Nit: here you're making it so the user will get an "unsupported model type" output for dvs that are either categorical or invariant. But the way you've written it, the "invariant" warning only works for non-numeric and non-boolean DVs. If a DV were e.g., c(1,1,1), you would be able run the model just fine, and you'd get all zero coefficients and hopefully you'd realize something was up and check your DV at that point. But, if you've been using this package a lot, it might be confusing that you didn't get the "invariant" warning that you would get for c("dog", "dog", "dog"). So I'd recommend either returning "invariant" here for all dvs with < 2 unique values, or else just have "categorical" be the only unsupported variable type.